elastic/beats
# libbeat/publisher/pipeline: expand monitoring

## What does this PR do?

Expand monitoring to report:

 - pipeline.queue.max_events (queue capacity)
 - output.batch_size (e.g. ES bulk_max_size)
 - output.clients (e.g. number of ES hosts * workers)

Queue capacity is recorded in stats rather than state so it can be used along with queue size to calculate saturation.

Note to reviewer, I introduced the `metricsObserverVars` type to avoid having awkward variable names just to avoid clashing with the method names.

## Why is it important?

In APM Server we recommend users tune `queue.mem.events`, `output.elasticsearch.bulk_max_size`, and `output.elasticsearch.worker` according to their scale. It would be helpful to be able to observe the current value in effect.

## Checklist

~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~

## How to test this PR locally

Run a beat with the Elasticsearch output and monitoring enabled, and check the values of `pipeline.queue.max_events`, `output.batch_size`, and `output.clients` match the configuration.

## Related issues

Closes #22215
